hetmech/bulk_computation.py

```diff
@@ -0,0 +1,58 @@
+import pandas
+import scipy.special
+import scipy.stats
+
+import hetmech.degree_group
+import hetmech.degree_weight
+import hetmech.hetmat
+
+
+def compute_save_dgp(hetmat, metapath, damping=0.5, compression='gzip', delete_intermediates=True):
+    """
+    Compute summary file of combined degree-grouped permutations (DGP). Aggregates across permutations,
+    deleting intermediates if delete_intermediates=True. Saves resulting files as compressed .tsv files
+    using compression method given by compression. Does not recompute previously saved files.
+    """
+    for mp in (metapath.inverse, metapath):
+        combined_path = hetmat.get_summary_degree_group_path(metapath, 'dwpc', damping)
+        if combined_path.exists():
+            return
+
+    _, _, matrix = hetmat.read_path_counts(metapath, 'dwpc', damping)
+    matrix_mean = matrix.mean()
+
+    for name, permat in hetmat.permutations.items():
+        path = permat.get_degree_group_path(metapath, 'dwpc', damping)
+        if not path.exists():
+            degree_grouped_df = hetmech.degree_group.single_permutation_degree_group(
+                permat, metapath, dwpc_mean=matrix_mean, damping=damping)
+            path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
I wonder whether we should do directory creation stuff in the path getters? Perhaps an optional `initialize` or `mkdir` arg?